pandas/core/generic.py

```diff
@@ -1130,8 +1130,8 @@ def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
         return packers.to_msgpack(path_or_buf, self, encoding=encoding,
                                   **kwargs)
 
-    def to_sql(self, name, con, flavor='sqlite', schema=None, if_exists='fail',
-               index=True, index_label=None, chunksize=None, dtype=None):
+    def to_sql(self, name, con, schema=None, if_exists='fail', index=True,
+               index_label=None, chunksize=None, dtype=None, *args, **kwargs):
```

Comment:
Alternative is to keep `flavor` in the signature, but change it to None, so we can detect when people set it to `'sqlite'` themselves. That solves the positional kwarg problem, but of course pollutes the signature a bit.